### ISSUE TITLE
[web:a11y] switch header from role "heading" to role "banner"

### ISSUE THREAD

**Issue by yjbanov** [flutter/flutter#152268]
On this line:

https://github.com/flutter/engine/blob/511a8ef6e69569419c54ec223eb20dd74e64eeed/lib/web_ui/lib/src/engine/semantics/semantics.dart#L715

When a semantic node self-identifies as `isHeader`, the web engine assigns the "heading" role. However, the more appropriate ARIA role would be "[banner](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/banner_role)".

It is also worth revisiting this line:

https://github.com/flutter/engine/blob/511a8ef6e69569419c54ec223eb20dd74e64eeed/lib/web_ui/lib/src/engine/semantics/semantics.dart#L690

Currently the `GenericRole` expects the header to have a label. However, the UI structure could be such that the header is a parent of multiple other semantic elements. See the [MDN example](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/header) that has a `<header>` containing an `<h1>` and `<time>`.

A more complete solution is probably to create a `PrimaryRoleManager` (to be renamed to `SemanticRole` in https://github.com/flutter/engine/pull/54070) for header, and stop relying on the `GenericRole`, which is really just a fallback when we couldn't think of a better role for a node.

**Comment by ditman** [comment]
Not all headers are banners, are they?

> By default, the HTML's [`<header>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/header) element has an identical meaning to the `banner` landmark, unless it is a descendant of [`<aside>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/aside), [`<article>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/article), [`<main>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/main), [`<nav>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/nav), or [`<section>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/section), at which point [`<header>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/header) exposes a [`generic`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/generic_role) role, and not the equivalent of the site-wide banner.

(From the [banner role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/banner_role) link)

**Comment by yjbanov** [comment]
@ditman 

I read that "a header/banner cannot be nested under aside/article/main/etc". "exposes a generic role" simply means it loses its header/banner role/meaning. It's like wrapping a `<h1>` in a `<button>`. It doesn't make sense any more. However, in the context where `<header>` makes sense, `role="banner"` also makes sense and vice versa, so we can use either for accessibility, and `<header>` both accessibility and indexing.

**Comment by math1man** [comment]
Regarding my point #2, here's the issue using Wikipedia as a proxy, given that it has a very similar layout to our app: 
![image](https://github.com/user-attachments/assets/7b42b93a-c515-45d0-afec-9f5fce8d6aae)

**Comment by yjbanov** [comment]
Reopening after reverting the fix.
